scispacy/custom_sentence_segmenter.py

```diff
@@ -31,32 +15,21 @@ def combined_rule_sentence_segmenter(doc: Doc) -> Doc:
 
     @param doc: the spaCy document to be annotated with sentence boundaries
     """
-    segmenter = pysbd.Segmenter(language='en', clean=False)
-    segments = segmenter.segment(doc.text)
-    segments = merge_segments(segments)
-
-    # pysbd splits raw text into sentences, so we have to do our best to align those
-    # segments with spacy tokens
-    segment_index = 0
-    current_segment = segments[segment_index]
-    built_up_sentence = ""
-    for i, token in enumerate(doc):
-        if i == 0 and (token.is_space or token.text == '.'):
-            token.is_sent_start = True
-            continue
-        if token.text.replace('\n', '').replace('\r', '') == '':
-            token.is_sent_start = False
-        elif len(built_up_sentence) >= len(current_segment):
-            token.is_sent_start = True
-
-            # handle the rare (impossible?) case where spacy tokenizes over a sentence boundary that
-            # pysbd finds
-            built_up_sentence = ' '*int(len(built_up_sentence) - len(current_segment))
-            built_up_sentence = token.text_with_ws
-            segment_index += 1
-            current_segment = segments[segment_index]
+    segmenter = pysbd.Segmenter(language="en", clean=False, char_span=True)
+    sents_char_spans = segmenter.segment(doc.text)
+
+    char_spans = [
```

Comment:
Yes using pysbd with spacy makes pysbd rely on spacy tokens. spacy requires `tok.is_sent_start` to be set so even if pysbd is able to give proper sentences with its character offsets, it depends on `doc.char_span(sent_span.start, sent_span.end)` to return the Span object. If it returns `None` then we get only one sentence instead of two.

Using only pysbd:
```python
import pysbd 
text = 'How about tomorrow?We can meet at eden garden.' 
seg = pysbd.Segmenter(language="en", clean=False) 
seg.segment(text)
# ['How about tomorrow?', 'We can meet at eden garden.']
```

Using pysbd as a spacy component:
```
doc = nlp(text)
combined_rule_sentence_segmenter(doc)
# ['How about tomorrow?We can meet at eden garden.']

Reply:
Adding `xfail` test for it

Reply:
:+1: